velox/vector/ComplexVector.cpp

```diff
@@ -809,31 +809,58 @@ VectorPtr RowVector::pushDictionaryToRowVectorLeaves(const VectorPtr& input) {
       wrappers, input->size(), input, input->pool());
 }
 
-void ArrayVectorBase::checkRanges() const {
-  std::unordered_map<vector_size_t, vector_size_t> seenElements;
-  seenElements.reserve(size());
+template <bool kHasNulls>
+vector_size_t ArrayVectorBase::nextNonEmpty(vector_size_t i) const {
+  while (i < size() &&
+         ((kHasNulls && bits::isBitNull(rawNulls(), i)) || rawSizes_[i] <= 0)) {
+    ++i;
+  }
+  return i;
+}
+
+template <bool kHasNulls>
+bool ArrayVectorBase::maybeHaveOverlappingRanges() const {
+  vector_size_t i = 0;
+  i = nextNonEmpty<kHasNulls>(i);
+  if (i >= size()) {
+    return false;
+  }
```

Comment:
How about
```
vector_size_t prev{-1};
for (;;) {
  const auto next = ...;
  ...
  if (prev != -1 && ...) {
    return false;
  }
  prev = next;
}
```

Reply:
Updated with `curr` and `next`.  Feels a little bit weird since these are usually used for pointers.

Reply:
Can you merge the code in l824 into the loop? curr{-1} can indicate the initial case? Thanks!

Reply:
There is call to `rawOffsets_[curr]` so `curr = -1` is not valid here